ibis/backends/dask/client.py

```diff
@@ -0,0 +1,342 @@
+"""The dask client implementation."""
+
+from __future__ import absolute_import
+
+import re
+from functools import partial
+
+import dask.dataframe as dd
+import dateutil.parser
+import numpy as np
+import pandas as pd
+import toolz
+from multipledispatch import Dispatcher
+from pandas.api.types import DatetimeTZDtype
+from pkg_resources import parse_version
+
+import ibis.client as client
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+import ibis.expr.types as ir
+from ibis.backends.pandas.client import (  # noqa: F401
+    PANDAS_DATE_TYPES,
+    PANDAS_STRING_TYPES,
+    _ibis_dtypes,
+    _inferable_pandas_dtypes,
+    _numpy_dtypes,
+    convert_timezone,
+    from_numpy_dtype,
+    from_pandas_categorical,
+    from_pandas_tzdtype,
+    ibis_dtype_to_pandas,
+    ibis_schema_to_pandas,
+    infer_array,
+    infer_numpy_scalar,
+    infer_pandas_timestamp,
+)
+
+from .core import execute_and_reset
+
+infer_dask_dtype = pd.api.types.infer_dtype
+
+
+_inferable_dask_dtypes = _inferable_pandas_dtypes
+
+
+@sch.schema.register(dd.Series)
+def schema_from_series(s):
+    return sch.schema(tuple(s.iteritems()))
+
+
+@sch.infer.register(dd.DataFrame)
+def infer_dask_schema(df, schema=None):
+    schema = schema if schema is not None else {}
+
+    pairs = []
+    for column_name, dask_dtype in df.dtypes.iteritems():
+        if not isinstance(column_name, str):
+            raise TypeError(
+                'Column names must be strings to use the dask backend'
+            )
+
+        if column_name in schema:
+            ibis_dtype = dt.dtype(schema[column_name])
+        elif dask_dtype == np.object_:
+            inferred_dtype = infer_dask_dtype(
+                df[column_name].compute(), skipna=True
+            )
+            if inferred_dtype in {'mixed', 'decimal'}:
+                # TODO: in principal we can handle decimal (added in pandas
+                # 0.23)
+                raise TypeError(
+                    'Unable to infer type of column {0!r}. Try instantiating '
+                    'your table from the client with client.table('
+                    "'my_table', schema={{{0!r}: <explicit type>}})".format(
+                        column_name
+                    )
+                )
+            ibis_dtype = _inferable_dask_dtypes[inferred_dtype]
+        else:
+            ibis_dtype = dt.dtype(dask_dtype)
+
+        pairs.append((column_name, ibis_dtype))
+
+    return sch.schema(pairs)
+
+
+ibis_dtype_to_dask = ibis_dtype_to_pandas
+
+ibis_schema_to_dask = ibis_schema_to_pandas
+
+convert = Dispatcher(
```

Comment:
Can you perhaps give this a better name? Looks like just this method is used to cast the type of dask Series?